backend/app/notify/email_test.go

```diff
@@ -0,0 +1,173 @@
+	req := request{comment: c}
+	mgs := email.buildMessageFromRequest(req, "test@localhost")
+	emptyTitleMessage := `From: noreply@example.org
+To: test@localhost
+Subject: New comment
+MIME-version: 1.0;
+Content-Type: text/html; charset="UTF-8";
+
+@from_user
+
+orig
+
+↦ <a href="//example.org#remark42__comment-">original comment</a>
+`
+	assert.Equal(t, emptyTitleMessage, mgs)
+	c.ParentID = "1"
+	c.PostTitle = "post title"
+	cp := store.Comment{Text: "some parent text"}
+	cp.User.Name = "@to_user"
+	req = request{comment: c, parent: cp}
+	filledTitleMessage := `From: noreply@example.org
+To: test@localhost
+Subject: New comment for "post title"
+MIME-version: 1.0;
+Content-Type: text/html; charset="UTF-8";
+
+@from_user → @to_user
+
+orig
+
+↦ <a href="//example.org#remark42__comment-">post title</a>
+`
+	mgs = email.buildMessageFromRequest(req, "test@localhost")
+	assert.Equal(t, filledTitleMessage, mgs)
+	// test sending
+	err = email.Send(context.Background(), req)
+	require.NoError(t, err)
+
+	assert.Equal(t, "noreply@example.org", fakeSMTP.mail)
+	assert.Equal(t, "test@localhost", fakeSMTP.rcpt)
+	assert.Equal(t, filledTitleMessage, fakeSMTP.buff.String())
+	assert.True(t, fakeSMTP.quit)
+	assert.False(t, fakeSMTP.close)
+}
+
+func TestBuildMessage(t *testing.T) {
+	email := Email{EmailParams: EmailParams{From: "from@email"}}
+	msg := email.buildMessage("test_subj", "test_body", "recepient@email", "")
+	expectedMsg := `From: from@email
+To: recepient@email
+Subject: test_subj
+
+test_body`
+	assert.Equal(t, expectedMsg, msg)
+	msg = email.buildMessage("test_subj", "test_body", "recepient@email", "text/html")
+	expectedLines := `MIME-version: 1.0;
+Content-Type: text/html; charset="UTF-8";`
+	assert.Contains(t, msg, expectedLines)
+}
+
+func TestConnectErrors(t *testing.T) {
+	email := Email{}
+	client, err := email.client()
+	assert.Nil(t, client)
+	assert.Error(t, err, "connection with wrong settings return error")
+	email = Email{EmailParams: EmailParams{TLS: true}}
+	client, err = email.client()
+	assert.Nil(t, client)
+	assert.Error(t, err, "TLS connection with wrong settings return error")
+}
+
+func TestEmailSendFailed(t *testing.T) {
+	fakeSMTP := &fakeTestSMTP{fail: true}
+	e := Email{EmailParams: EmailParams{From: "from@example.com"}, SMTPClient: fakeSMTP}
+	err := e.sendEmail("some text", "to@example.com")
+	require.EqualError(t, err, "can't make email writer: failed")
+
+	assert.Equal(t, "from@example.com", fakeSMTP.mail)
+	assert.Equal(t, "to@example.com", fakeSMTP.rcpt)
+	assert.Equal(t, "", fakeSMTP.buff.String())
+	assert.False(t, fakeSMTP.quit)
+	assert.True(t, fakeSMTP.close)
+}
+
+func TestEmailMultipleSend(t *testing.T) {
+	fakeSMTP := &fakeTestSMTP{}
+	waitCh := make(chan int)
+	var waitGroup sync.WaitGroup
+	e := Email{EmailParams: EmailParams{}, SMTPClient: fakeSMTP}
+	for i := 1; i <= 10; i++ {
+		waitGroup.Add(1)
+		go func() {
+			// will start once we close the channel
+			<-waitCh
+			_ = e.sendEmail("", "")
```

Comment:
this is the test, skipping error check here is not the best idea. assert.NoError(t, e.sendEmail) feels better even it makes a little sense with fake agent